File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/aboutme/AboutMeFragment.java
package com.lucagiorgetti.surprix.ui.mainfragments.aboutme;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.SearchView;
import android.widget.TextView;

import com.lucagiorgetti.surprix.R;
import com.lucagiorgetti.surprix.SurprixApplication;
import com.lucagiorgetti.surprix.model.User;
import com.lucagiorgetti.surprix.ui.activities.LoginActivity;
import com.lucagiorgetti.surprix.ui.activities.SettingsActivity;
import com.lucagiorgetti.surprix.ui.mainfragments.catalog.CatalogNavigationMode;
import com.lucagiorgetti.surprix.ui.mainfragments.catalog.producers.CatalogProducerFragmentDirections;
import com.lucagiorgetti.surprix.utility.BaseFragment;
import com.lucagiorgetti.surprix.utility.SystemUtils;

public class AboutMeFragment extends BaseFragment {
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        User user = SurprixApplication.getInstance().getCurrentUser();

        View root = inflater.inflate(R.layout.fragment_about_me, container, false);

        View collectionBtn = root.findViewById(R.id.my_collection_btn);
        Button logout = root.findViewById(R.id.logout_btn);

        TextView username = root.findViewById(R.id.username);
        TextView email = root.findViewById(R.id.email);

        username.setText(user.getUsername());
        email.setText(user.getEmail());

        collectionBtn.setOnClickListener(view -> {
            NavDirections action = AboutMeFragmentDirections.actionNavigationAboutMeToNavigationMyCollection();
            Navigation.findNavController(view).navigate(action);
        });

        logout.setOnClickListener(v -> {
            SystemUtils.logout();
            SystemUtils.openNewActivityWithFinishing(getActivity(), LoginActivity.class);
        });

        setHasOptionsMenu(true);
        return root;
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.my_collection_menu, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.nav_settings) {
            SystemUtils.openNewActivity(SettingsActivity.class, null);
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/aboutme/AboutMeFragment.kt
package com.lucagiorgetti.surprix.ui.mainfragments.aboutme

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.core.view.MenuHost
import androidx.core.view.MenuProvider
import androidx.lifecycle.Lifecycle
import androidx.navigation.Navigation.findNavController
import com.lucagiorgetti.surprix.R
import com.lucagiorgetti.surprix.SurprixApplication
import com.lucagiorgetti.surprix.model.User
import com.lucagiorgetti.surprix.ui.activities.LoginActivity
import com.lucagiorgetti.surprix.ui.activities.SettingsActivity
import com.lucagiorgetti.surprix.utility.BaseFragment
import com.lucagiorgetti.surprix.utility.SystemUtils

class AboutMeFragment : BaseFragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val user: User = SurprixApplication.instance.currentUser!!
        val root = inflater.inflate(R.layout.fragment_about_me, container, false)
        val collectionBtn = root.findViewById<View>(R.id.my_collection_btn)
        val logout = root.findViewById<Button>(R.id.logout_btn)
        val username = root.findViewById<TextView>(R.id.username)
        val email = root.findViewById<TextView>(R.id.email)
        username.text = user.username
        email.text = user.clearedEmail()
        collectionBtn.setOnClickListener {
            val action = AboutMeFragmentDirections.actionNavigationAboutMeToNavigationMyCollection()
            findNavController(requireView()).navigate(action)
        }
        logout.setOnClickListener {
            SystemUtils.logout()
            SystemUtils.openNewActivityWithFinishing(requireActivity(), LoginActivity::class.java)
        }

        val menuHost:MenuHost = requireActivity()
        menuHost.addMenuProvider(object: MenuProvider {
            override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
                menuInflater.inflate(R.menu.my_collection_menu, menu)
            }

            override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
                return when (menuItem.itemId) {
                    R.id.nav_settings -> {
                        SystemUtils.openNewActivity(SettingsActivity::class.java, null)
                        true
                    }
                    else -> false
                }
            }

        }, viewLifecycleOwner, Lifecycle.State.RESUMED)

        return root
    }
}